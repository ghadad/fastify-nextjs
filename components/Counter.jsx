import React from "react";

const Counter = ({ startingCount = 0 }) => {
  const [count, setCount] = React.useState(startingCount);
  return (
    <React.Fragment>
      <button className="incr" onClick={() => setCount(count + 1)}>
        +
      </button>
      <button className="decr" onClick={() => setCount(count - 1)}>
        -
      </button>
      <p className="count">{count}</p>
    </React.Fragment>
  );
};

export default Counter;
