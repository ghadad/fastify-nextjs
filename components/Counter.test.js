/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent, waitForElement } from 'react-testing-library'

import Counter from './Counter.jsx'

const renderComponent = ({ startingCount }) => (
  render(<Counter startingCount={startingCount} />)
)

describe('With React Testing Library', () => {
  it('renders initial count', async () => {
    const { getByText } = renderComponent({ startingCount: 5 })
    await waitForElement(() => getByText(/5/))
  })
  it('increments count', async () => {
    const { getByText } = renderComponent({ startingCount: 0 })
    await waitForElement(() => getByText('+'))
    fireEvent.click(getByText('+'))
    await waitForElement(() => getByText(/1/))
  })

  it('decrements count', async () => {
    const { getByText } = renderComponent({ startingCount: 3 })
    await waitForElement(() => getByText('-'))
    fireEvent.click(getByText('-'))
    await waitForElement(() => getByText(/2/))
  })
})

describe('With Snapshot Testing', () => {
  it('Counter shows up', () => {
    const component = renderer.create(<Counter />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
