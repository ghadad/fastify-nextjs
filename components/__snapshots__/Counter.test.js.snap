// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Counter shows up 1`] = `
Array [
  <button
    className="incr"
    onClick={[Function]}
  >
    +
  </button>,
  <button
    className="decr"
    onClick={[Function]}
  >
    -
  </button>,
  <p
    className="count"
  >
    0
  </p>,
]
`;
